src/haxesharp/text/Regex.hx

```diff
@@ -75,7 +75,27 @@ class Regex
     */
     public function match(input:String):Match
     {                
-        var groups = Regex.matches(input, this.regex, this.options);
+        var groups = Regex.matches(input, this.regex, this.options);        
         return new Match(groups.length > 0, groups);
     }
+
+    /**
+    Finds all instances where this regex matches the input string, and
+    replaces those with the replacement string. Does not (yet) support
+    groups or named groups.
+    **/
+    public function replace(input:String, replacement:String):String

```

Comment:
`replacement` variable is unused, which means something like `.replace(myRegex, "test")` is just removing any matches of `myRegex` without substituting in the `replacement` string.

Reply:
This is the main use case I was trying to replace (remove comments in a JSON file). I totally missed this. I'll add a test case too. Good catch!